tracks/go/exercises/nucleotide-count/mentoring.md

```diff
@@ -0,0 +1,98 @@
+# Nucleotide Count
+
+### Reasonable Solutions
+
+Using a map is probably the most idiomatic solution to this problem, especially since the Histogram type basically must be a map. Using a switch is about 4 times faster over the provided test cases because it does not require hashing.
+
+#### Map Example
+```Go
+// Package dna counts nucleotides
+package dna
+
+import "errors"
+
+// Histogram is a mapping from nucleotide to its count in given DNA.
+type Histogram map[rune]int
+
+// DNA is a list of nucleotides.
+type DNA string
+
+// Counts generates a histogram of valid nucleotides in the given DNA.
+// Returns an error if d contains an invalid nucleotide.
+func (d DNA) Counts() (Histogram, error) {
+	h := Histogram{
+		'A': 0,
+		'C': 0,
+		'G': 0,
+		'T': 0,
+	}
+	for _, r := range d {
+		if _, ok := h[r]; !ok {
+			return nil, errors.New("invalid nucleotide")
+		}
+		h[r]++
+	}
+	return h, nil
+}
+```
+
+#### Switch Example
+```Go
+// Package dna counts nucleotides
+package dna
+
+import "errors"
+
+// Histogram is a mapping from nucleotide to its count in given DNA.
+type Histogram map[rune]int
+
+// DNA is a list of nucleotides.
+type DNA string
+
+// Nucleotide indexes for counting
+const (
+	A = iota
+	C
+	G
+	T
+)
+
+// ErrInvalidNucleotide is returned when it encounters something other than A/C/G/T
+var ErrInvalidNucleotide = errors.New("invalid nucleotide")
+
+// Counts generates a histogram of valid nucleotides in the given DNA.
+// Returns an error if d contains an invalid nucleotide.
+func (d DNA) Counts() (Histogram, error) {
+	counts := [4]int{}
+	for _, nucleotide := range d {
+		switch nucleotide {
+		case 'A':
+			counts[A]++
+		case 'C':
+			counts[C]++
+		case 'G':
+			counts[G]++
+		case 'T':
+			counts[T]++
+		default:
+			return nil, ErrInvalidNucleotide
+		}
+	}
+	return Histogram{
+		'A': counts[A],
+		'C': counts[C],
+		'G': counts[G],
+		'T': counts[T],
+	}, nil
+}
+```
+
+### Common Suggestions
+- When you have a type definition for the Histogram, you can use it like the underlying type. For instance, you can `make(Histogram, 4)` or instantiate a literal `Histogram{...}`.
```

Comment:
> for the Histogram

When your define a Histogram type?

Reply:
I was trying to be consistent with the specific language of [Type Definitions](https://go.dev/ref/spec#Type_definitions), but that probably doesn't matter as much as it seemed like it might at the time I wrote that.